x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java

```diff
@@ -595,6 +609,7 @@ public long physicalEndOffset() {
         public boolean tryEvict() {
             if (refCount() <= 1 && evicted.compareAndSet(false, true)) {
                 logger.trace("evicted {} with channel offset {}", regionKey, physicalStartOffset());
+                evictCount.increment(); // only count evictions that are not related to shard shutdown
```

Comment:
why not count those?

Reply:
Somehow I had the impression that we considered the evictions from shard removal/relocation as part of the normal lifecycle of shards and not interesting in the context of cache statistics.

I pushed aceb3ac215aefb5470e3772e7b1f696a97d75630 to count all eviction. Let me know what you think.